polyfill/lib/datetime.mjs

```diff
@@ -233,16 +233,13 @@ export class DateTime {
     let year = GetSlot(date, ISO_YEAR);
     let month = GetSlot(date, ISO_MONTH);
     let day = GetSlot(date, ISO_DAY);
-    let { hour, minute, second, millisecond, microsecond, nanosecond } = fields;
-    ({ hour, minute, second, millisecond, microsecond, nanosecond } = ES.RegulateTime(
-      hour,
-      minute,
-      second,
-      millisecond,
-      microsecond,
-      nanosecond,
-      overflow
-    ));
+    const time = calendar.timeFromFields(fields, options, GetIntrinsic('%Temporal.Time%'));
```

Comment:
```suggestion
    const time = calendar.timeFromFields(fields, { overflow }, GetIntrinsic('%Temporal.Time%'));
```
I think we are normally creating a new options object to pass to the calendar methods instead of passing through the user-supplied one.

Reply:
@ptomato check the `dateFromFields` call *right* above this. It passes in `options` directly. I spotted a lot of such issues early-on and even made a few refactor commits that I later dropped to keep this PR as small and on-topic as possible. Should I fix all sites in this PR, or would a single follow-up that takes care of all such cases suffice?